File: heart/src/main/java/com/vantagecircle/heartrate/AppComponent.java
package com.vantagecircle.heartrate;

import android.app.Application;
import android.content.Context;

import com.vantagecircle.heartrate.data.DataManager;
import com.vantagecircle.heartrate.data.DatabaseHelper;
import com.vantagecircle.heartrate.scope.ApplicationContext;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by bapidas on 09/10/17.
 */
@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(HeartApplication heartApplication);
    @ApplicationContext
    Context getContext();
    Application getApplication();
    DataManager getDataManager();
    DatabaseHelper getDatabaseHelper();
}

File: app/src/main/java/com/bapidas/heartrate/di/component/AppComponent.kt
package com.bapidas.heartrate.di.component

import android.app.Application
import android.content.Context
import com.bapidas.heartrate.HeartApplication
import com.bapidas.heartrate.data.DataManager
import com.bapidas.heartrate.di.annotation.ApplicationContext
import com.bapidas.heartrate.di.module.AppModule
import com.bapidas.heartrate.di.module.DatabaseModule
import dagger.Component

/**
 * Created by bapidas on 09/10/17.
 */
@Component(modules = [AppModule::class, DatabaseModule::class])
interface AppComponent {
    fun inject(heartApplication: HeartApplication?)

    @get:ApplicationContext
    val context: Context

    val application: Application

    val dataManager: DataManager
}